File: src/main/java/com/spring/app/orders/payload/OrderItemData.java
package com.spring.app.orders.payload;

import com.spring.app.products.payload.PackageData;
import com.spring.app.shipping.payload.ShippingAddressData;

import java.util.List;

public class OrderItemData {
    private Long id;
    private Double price;
    private ShippingAddressData shippingAddress;
    private List<PackageData> products;

    public OrderItemData() {}

    public OrderItemData(
            Long id,
            Double price,
            ShippingAddressData shippingAddress,
            List<PackageData> products
    ) {
        this.id = id;
        this.price = price;
        this.shippingAddress = shippingAddress;
        this.products = products;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public ShippingAddressData getShippingAddress() {
        return shippingAddress;
    }

    public void setShippingAddress(ShippingAddressData shippingAddress) {
        this.shippingAddress = shippingAddress;
    }

    public List<PackageData> getProducts() {
        return products;
    }

    public void setProducts(List<PackageData> products) {
        this.products = products;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }
}

File: views/src/components/models/order/order-item-data.js
import ShippingData from "@/components/models/shipping/shipping-data";

export default class OrderItemData {
    shippingData = new ShippingData();

    data = {
        id: null,
        price: 0.0,
        shippingAddress: this.shippingData.getData(),
        products: []
    };

    setData (data) {
        this.data.id = data.id;
        this.data.price = data.price;
        this.data.shippingAddress = data.shippingAddress;
        this.data.products = data.products;
    }

    getData () {
        return this.data;
    }

    setShippingAddress(shippingAddress) {
        this.data.shippingAddress = shippingAddress;
    }

    setProduct(products) {
        this.data.products = products;
    }
}